### DISCUSSION TITLE
Reduce startup time

### DISCUSSION THREAD

**Discussion by maveduck** [glowingjade/obsidian-smart-composer#102] [Ideas & Feature Requests]
First of all I love this plugin. I tried a lot of AI plugins but this is the one I use. Thank you for that.

As a feature request I would like the startup time to be shorter. Right now it is 350ms in my desktop app on Mac OS. Maybe it would be possible to lazy load more things to speed it up? 

<img width="488" alt="image" src="https://github.com/user-attachments/assets/0c9fa72a-0d96-444d-a2bf-06cfb6378a1e">

**Comment by glowingjade** [comment]
Hi! Thanks for the suggestion. I'll look into what we can do to reduce startup time.

**Comment by shayonpal** [comment]
> <img alt="image" width="488" src="https://private-user-images.githubusercontent.com/123307518/385955442-0c9fa72a-0d96-444d-a2bf-06cfb6378a1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxODczMDUsIm5iZiI6MTczNjE4NzAwNSwicGF0aCI6Ii8xMjMzMDc1MTgvMzg1OTU1NDQyLTBjOWZhNzJhLTBkOTYtNDQ0ZC1hMmJmLTA2Y2ZiNjM3OGExZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNlQxODEwMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYzY4MzY2ZGRjYzJlMTlmYjE2MDk1ZDRhYTYyOGVkYWEzZWUyYTgzYzg1YTgxMGQ1NTlmNTZiZWYzMjE2MDZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.eAbj8FI_Fu_L1rLiNAr1xANeG5PqnbcaE0aIg_j3JFk">

How does one get this startup time report?

**Comment by glowingjade** [comment]
<img width="1094" alt="Screenshot 2025-01-07 at 10 43 36 AM" src="https://github.com/user-attachments/assets/60225c4b-c31c-462f-9c94-d1b6b4dcbc30" />
@shayonpal  You can view it by clicking the timer icon in General > Advanced menu in settings.
